delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/ResourceSourceParser.scala

```diff
@@ -34,8 +82,7 @@ trait ResourceSourceParser {
   ): IO[InvalidJsonLdFormat, (Iri, CompactedJsonLd, ExpandedJsonLd)] =
     for {
       (ctx, originalExpanded) <- expandSource(project, source)
-      base                     = project.base.iri
-      iri                     <- originalExpanded.rootId.asIri.fold(uuidF().map(uuid => base / uuid.toString))(IO.pure)
+      iri                     <- getOrGenerateId(originalExpanded.rootId.asIri, project)
       expanded                 = originalExpanded.replaceId(iri)
       compacted               <- expanded.toCompacted(ctx).leftMap(err => InvalidJsonLdFormat(Some(iri), err))
```

Comment:
Same here. Why this is fixed to a `ResourceRejection.InvalidJsonLdFormat`. A resolver can issue its own `ResolverRejection.InvalidJsonLdFormat` since it is not the responsibility of Resources to check for this